dom/src/main/scala/fs2/dom/AnimationEvent.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import cats.effect.kernel.Sync
+import org.scalajs.dom
+
+abstract class AnimationEvent[F[_]] private[dom] extends Event[F] {}
+
+object AnimationEvent {
+  def apply[F[_]](event: dom.AnimationEvent)(implicit F: Sync[F]): AnimationEvent[F] =
+    new WrappedAnimationEvent(event)
```

Comment:
I'm curious why not using anonymous class for implementing `AnimationEvent` right here? Would save a few lines, not just for AnimationEvent.

Reply:
Hm you're right 😁 don't remember if I had a good reason, it took a few tries to figure out the inheritance structure to share implementation.

Reply:
Oh, it's because it becomes a bit annoying, like this:
```scala
  def apply[F[_]](ev: dom.AnimationEvent)(implicit ev: Sync[F]): AnimationEvent[F] =
    new AnimationEventImpl {
      val event = ev
      val F = ev
    }
```